### ISSUE TITLE
LLVM error: mismatched subprogram between llvm.dbg.declare variable and !dbg attachment

### ISSUE THREAD

**Issue by jgreener64** [EnzymeAD/Enzyme.jl#533]
I am on Enzyme main (d262d136d937216c13e2434c5a32b14183ed5159), CUDA 3.12.0 and Julia 1.8.2. The following errors:
```julia
using Enzyme, CUDA

function kernel!(a)
    if threadIdx().x == 1
        a[1] = 1.0f0
    end
    return nothing
end

function grad_kernel!(a, da)
    Enzyme.autodiff_deferred(
        kernel!,
        Duplicated(a, da),
    )
    return nothing
end

a = CUDA.rand(256)
da = zero(a)

CUDA.@sync @cuda threads=1 blocks=1 grad_kernel!(a, da)
```
```
signal (11): Segmentation fault
in expression starting at REPL[2]:1
_ZN4llvm10DwarfDebug23emitInitialLocDirectiveERKNS_15MachineFunctionEj at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm10AsmPrinter31emitInitialRawDwarfLocDirectiveERKNS_15MachineFunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm15NVPTXAsmPrinter22emitFunctionEntryLabelEv at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm10AsmPrinter18emitFunctionHeaderEv at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm10AsmPrinter16emitFunctionBodyEv at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm15NVPTXAsmPrinter20runOnMachineFunctionERNS_15MachineFunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm19MachineFunctionPass13runOnFunctionERNS_8FunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZL21LLVMTargetMachineEmitP23LLVMOpaqueTargetMachineP16LLVMOpaqueModuleRN4llvm17raw_pwrite_streamE19LLVMCodeGenFileTypePPc at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
LLVMTargetMachineEmitToMemoryBuffer at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
LLVMTargetMachineEmitToMemoryBuffer at /home/jgreener/.julia/packages/LLVM/WjSQG/lib/13/libLLVM_h.jl:947 [inlined]
emit at /home/jgreener/.julia/packages/LLVM/WjSQG/src/targetmachine.jl:45
mcgen at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/mcgen.jl:73
unknown function (ip: 0x7fadf15efcdf)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
macro expansion at /home/jgreener/.julia/packages/TimerOutputs/4yHI4/src/TimerOutput.jl:253 [inlined]
macro expansion at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/driver.jl:430 [inlined]
macro expansion at /home/jgreener/.julia/packages/TimerOutputs/4yHI4/src/TimerOutput.jl:253 [inlined]
macro expansion at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/driver.jl:427 [inlined]
#emit_asm#118 at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/utils.jl:68
unknown function (ip: 0x7fadf15e7f7d)
emit_asm##kw at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/utils.jl:62 [inlined]
cufunction_compile at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:354
#224 at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:347 [inlined]
JuliaContext at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/driver.jl:76
unknown function (ip: 0x7fadf1621c5a)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
cufunction_compile at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:346
cached_compilation at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/cache.jl:90
#cufunction#221 at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:299
cufunction at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:292
unknown function (ip: 0x7fadf16215bf)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
jl_apply at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/julia.h:1839 [inlined]
do_call at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:126
eval_value at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:215
eval_body at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:467
jl_interpret_toplevel_thunk at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:750
jl_toplevel_eval_flex at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/toplevel.c:906
jl_toplevel_eval_flex at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/toplevel.c:850
ijl_toplevel_eval_in at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/toplevel.c:965
eval at ./boot.jl:368 [inlined]
eval_user_input at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:151
repl_backend_loop at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:247
start_repl_backend at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:232
#run_repl#47 at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:369
run_repl at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:355
jfptr_run_repl_64841.clone_1 at /home/jgreener/soft/julia/julia-1.8.2/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
#967 at ./client.jl:419
jfptr_YY.967_30403.clone_1 at /home/jgreener/soft/julia/julia-1.8.2/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
jl_apply at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/julia.h:1839 [inlined]
jl_f__call_latest at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/builtins.c:774
#invokelatest#2 at ./essentials.jl:729 [inlined]
invokelatest at ./essentials.jl:726 [inlined]
run_main_repl at ./client.jl:404
exec_options at ./client.jl:318
_start at ./client.jl:522
jfptr__start_56736.clone_1 at /home/jgreener/soft/julia/julia-1.8.2/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
jl_apply at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/julia.h:1839 [inlined]
true_main at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/jlapi.c:575
jl_repl_entrypoint at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/jlapi.c:719
main at julia (unknown line)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x401098)
Allocations: 62987151 (Pool: 62935186; Big: 51965); GC: 61
Segmentation fault (core dumped)
```
Output with `Enzyme.API.printall!(true)`:
```
after simplification :
; Function Attrs: mustprogress willreturn
define void @preprocess_julia_kernel__5023_inner7({ i8 addrspace(1)*, i64, [1 x i64], i64 } %0) local_unnamed_addr #8 !dbg !167 {
entry:
  %.fca.0.extract = extractvalue { i8 addrspace(1)*, i64, [1 x i64], i64 } %0, 0, !dbg !168
  %.fca.2.0.extract = extractvalue { i8 addrspace(1)*, i64, [1 x i64], i64 } %0, 2, 0, !dbg !168
  %1 = call {}*** @julia.get_pgcstack() #9
  %2 = call i32 @llvm.nvvm.read.ptx.sreg.tid.x() #9, !dbg !169, !range !113
  %.not = icmp eq i32 %2, 0, !dbg !176
  %3 = bitcast i8 addrspace(1)* %.fca.0.extract to float addrspace(1)*, !dbg !178
  br i1 %.not, label %L14.i, label %julia_kernel__5023_inner.exit, !dbg !178

L14.i:                                            ; preds = %entry
  %4 = icmp slt i64 %.fca.2.0.extract, 1, !dbg !179
  br i1 %4, label %L27.i, label %L25.i, !dbg !193

L25.i:                                            ; preds = %L14.i
  store float 1.000000e+00, float addrspace(1)* %3, align 4, !dbg !194, !tbaa !162
  br label %julia_kernel__5023_inner.exit, !dbg !200

L27.i:                                            ; preds = %L14.i
  %5 = call fastcc nonnull {} addrspace(10)* @julia__throw_boundserror_5030() #10, !dbg !193
  unreachable, !dbg !193

julia_kernel__5023_inner.exit:                    ; preds = %L25.i, %entry
  ret void, !dbg !168
}

; Function Attrs: mustprogress willreturn
define internal void @diffejulia_kernel__5023_inner7({ i8 addrspace(1)*, i64, [1 x i64], i64 } %0, { i8 addrspace(1)*, i64, [1 x i64], i64 } %"'") local_unnamed_addr #8 !dbg !201 {
entry:
  %".fca.0.extract'ipev" = extractvalue { i8 addrspace(1)*, i64, [1 x i64], i64 } %"'", 0, !dbg !202
  %.fca.0.extract = extractvalue { i8 addrspace(1)*, i64, [1 x i64], i64 } %0, 0, !dbg !202
  %.fca.2.0.extract = extractvalue { i8 addrspace(1)*, i64, [1 x i64], i64 } %0, 2, 0, !dbg !202
  %1 = call {}*** @julia.get_pgcstack() #9
  %2 = call i32 @llvm.nvvm.read.ptx.sreg.tid.x() #9, !dbg !203, !range !113
  %.not = icmp eq i32 %2, 0, !dbg !210
  %"'ipc" = bitcast i8 addrspace(1)* %".fca.0.extract'ipev" to float addrspace(1)*, !dbg !212
  %3 = bitcast i8 addrspace(1)* %.fca.0.extract to float addrspace(1)*, !dbg !212
  br i1 %.not, label %L14.i, label %julia_kernel__5023_inner.exit, !dbg !212

L14.i:                                            ; preds = %entry
  %4 = icmp slt i64 %.fca.2.0.extract, 1, !dbg !213
  br i1 %4, label %L27.i, label %L25.i, !dbg !227

L25.i:                                            ; preds = %L14.i
  store float 1.000000e+00, float addrspace(1)* %3, align 4, !dbg !228, !tbaa !162, !alias.scope !234, !noalias !237
  br label %julia_kernel__5023_inner.exit, !dbg !239

L27.i:                                            ; preds = %L14.i
  %5 = call fastcc nonnull {} addrspace(10)* @julia__throw_boundserror_5030() #10, !dbg !227
  unreachable

julia_kernel__5023_inner.exit:                    ; preds = %L25.i, %entry
  br label %invertjulia_kernel__5023_inner.exit, !dbg !202

invertentry:                                      ; preds = %invertjulia_kernel__5023_inner.exit, %invertL14.i
  ret void

invertL14.i:                                      ; preds = %invertL25.i
  br label %invertentry

invertL25.i:                                      ; preds = %invertjulia_kernel__5023_inner.exit
  store float 0.000000e+00, float addrspace(1)* %"'ipc", align 4, !dbg !228, !tbaa !162, !alias.scope !237, !noalias !234
  br label %invertL14.i

invertjulia_kernel__5023_inner.exit:              ; preds = %julia_kernel__5023_inner.exit
  br i1 %.not, label %invertL25.i, label %invertentry
}


signal (11): Segmentation fault
in expression starting at REPL[1]:1
_ZN4llvm10DwarfDebug23emitInitialLocDirectiveERKNS_15MachineFunctionEj at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm10AsmPrinter31emitInitialRawDwarfLocDirectiveERKNS_15MachineFunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm15NVPTXAsmPrinter22emitFunctionEntryLabelEv at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm10AsmPrinter18emitFunctionHeaderEv at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm10AsmPrinter16emitFunctionBodyEv at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm15NVPTXAsmPrinter20runOnMachineFunctionERNS_15MachineFunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm19MachineFunctionPass13runOnFunctionERNS_8FunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
_ZL21LLVMTargetMachineEmitP23LLVMOpaqueTargetMachineP16LLVMOpaqueModuleRN4llvm17raw_pwrite_streamE19LLVMCodeGenFileTypePPc at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
LLVMTargetMachineEmitToMemoryBuffer at /home/jgreener/soft/julia/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
LLVMTargetMachineEmitToMemoryBuffer at /home/jgreener/.julia/packages/LLVM/WjSQG/lib/13/libLLVM_h.jl:947 [inlined]
emit at /home/jgreener/.julia/packages/LLVM/WjSQG/src/targetmachine.jl:45
mcgen at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/mcgen.jl:73
unknown function (ip: 0x7f19ec2a2fbf)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
macro expansion at /home/jgreener/.julia/packages/TimerOutputs/4yHI4/src/TimerOutput.jl:253 [inlined]
macro expansion at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/driver.jl:430 [inlined]
macro expansion at /home/jgreener/.julia/packages/TimerOutputs/4yHI4/src/TimerOutput.jl:253 [inlined]
macro expansion at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/driver.jl:427 [inlined]
#emit_asm#118 at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/utils.jl:68
unknown function (ip: 0x7f19ec29b25d)
emit_asm##kw at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/utils.jl:62 [inlined]
cufunction_compile at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:354
#224 at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:347 [inlined]
JuliaContext at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/driver.jl:76
unknown function (ip: 0x7f19ec2ceb0a)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
cufunction_compile at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:346
cached_compilation at /home/jgreener/.julia/packages/GPUCompiler/07qaN/src/cache.jl:90
#cufunction#221 at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:299
cufunction at /home/jgreener/.julia/packages/CUDA/DfvRa/src/compiler/execution.jl:292
unknown function (ip: 0x7f19ec2ce46f)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
jl_apply at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/julia.h:1839 [inlined]
do_call at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:126
eval_value at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:215
eval_body at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:467
jl_interpret_toplevel_thunk at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/interpreter.c:750
jl_toplevel_eval_flex at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/toplevel.c:906
jl_toplevel_eval_flex at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/toplevel.c:850
ijl_toplevel_eval_in at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/toplevel.c:965
eval at ./boot.jl:368 [inlined]
eval_user_input at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:151
repl_backend_loop at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:247
start_repl_backend at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:232
#run_repl#47 at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:369
run_repl at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/REPL/src/REPL.jl:355
jfptr_run_repl_64841.clone_1 at /home/jgreener/soft/julia/julia-1.8.2/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
#967 at ./client.jl:419
jfptr_YY.967_30403.clone_1 at /home/jgreener/soft/julia/julia-1.8.2/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
jl_apply at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/julia.h:1839 [inlined]
jl_f__call_latest at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/builtins.c:774
#invokelatest#2 at ./essentials.jl:729 [inlined]
invokelatest at ./essentials.jl:726 [inlined]
run_main_repl at ./client.jl:404
exec_options at ./client.jl:318
_start at ./client.jl:522
jfptr__start_56736.clone_1 at /home/jgreener/soft/julia/julia-1.8.2/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2367 [inlined]
ijl_apply_generic at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/gf.c:2549
jl_apply at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/julia.h:1839 [inlined]
true_main at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/jlapi.c:575
jl_repl_entrypoint at /cache/build/default-amdci4-6/julialang/julia-release-1-dot-8/src/jlapi.c:719
main at julia (unknown line)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x401098)
Allocations: 61397859 (Pool: 61346623; Big: 51236); GC: 61
Segmentation fault (core dumped)
```
CUDA version info:
```
CUDA toolkit 11.7, artifact installation
NVIDIA driver 470.141.3, for CUDA 11.4
CUDA driver 11.7

Libraries: 
- CUBLAS: 11.10.1
- CURAND: 10.2.10
- CUFFT: 10.7.2
- CUSOLVER: 11.3.5
- CUSPARSE: 11.7.3
- CUPTI: 17.0.0
- NVML: 11.0.0+470.141.3
- CUDNN: 8.30.2 (for CUDA 11.5.0)
- CUTENSOR: 1.4.0 (for CUDA 11.5.0)

Toolchain:
- Julia: 1.8.2
- LLVM: 13.0.1
- PTX ISA support: 3.2, 4.0, 4.1, 4.2, 4.3, 5.0, 6.0, 6.1, 6.3, 6.4, 6.5, 7.0, 7.1, 7.2
- Device capability support: sm_35, sm_37, sm_50, sm_52, sm_53, sm_60, sm_61, sm_62, sm_70, sm_72, sm_75, sm_80, sm_86

2 devices:
  0: NVIDIA RTX A6000 (sm_86, 47.531 GiB / 47.544 GiB available)
  1: NVIDIA RTX A6000 (sm_86, 42.360 GiB / 47.541 GiB available)
```

**Comment by wsmoses** [comment]
```

ERROR: LLVM error: mismatched subprogram between llvm.dbg.declare variable and !dbg attachment
  call void @llvm.dbg.declare(metadata { i8 addrspace(1)*, i64, [1 x i64], i64 }* undef, metadata !239, metadata !DIExpression(DW_OP_deref)), !dbg !244
label %conversion
void ([1 x i64], { i8 addrspace(1)*, i64, [1 x i64], i64 }, { i8 addrspace(1)*, i64, [1 x i64], i64 })* @_Z23julia_grad_kernel__358513CuDeviceArrayI7Float32Li1ELi1EES_IS0_Li1ELi1EE
!239 = !DILocalVariable(name: "a", arg: 2, scope: !240, file: !4, line: 1, type: !92)
!240 = distinct !DISubprogram(name: "grad_kernel!", linkageName: "julia_grad_kernel!_3585", scope: null, file: !4, line: 1, type: !232, scopeLine: 1, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !3, retainedNodes: !241)
!244 = !DILocation(line: 1, scope: !231)
!231 = distinct !DISubprogram(name: "grad_kernel!", linkageName: "julia_grad_kernel!_3585", scope: null, file: !4, line: 1, type: !232, scopeLine: 1, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !3, retainedNodes: !235)
mismatched subprogram between llvm.dbg.declare variable and !dbg attachment
  call void @llvm.dbg.declare(metadata { i8 addrspace(1)*, i64, [1 x i64], i64 }* undef, metadata !243, metadata !DIExpression(DW_OP_deref)), !dbg !244
label %conversion
void ([1 x i64], { i8 addrspace(1)*, i64, [1 x i64], i64 }, { i8 addrspace(1)*, i64, [1 x i64], i64 })* @_Z23julia_grad_kernel__358513CuDeviceArrayI7Float32Li1ELi1EES_IS0_Li1ELi1EE
!243 = !DILocalVariable(name: "da", arg: 3, scope: !240, file: !4, line: 1, type: !92)
!240 = distinct !DISubprogram(name: "grad_kernel!", linkageName: "julia_grad_kernel!_3585", scope: null, file: !4, line: 1, type: !232, scopeLine: 1, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !3, retainedNodes: !241)
!244 = !DILocation(line: 1, scope: !231)
!231 = distinct !DISubprogram(name: "grad_kernel!", linkageName: "julia_grad_kernel!_3585", scope: null, file: !4, line: 1, type: !232, scopeLine: 1, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !3, retainedNodes: !235)
```

@vchuravy

**Comment by jgreener64** [comment]
Is there an estimate on time frame for fixing this issue?

**Comment by vchuravy** [comment]
Sorry, you hit the conference travel season. Hopefully I will find some time next week. 

Just to clarify does this happen when you start Julia with `-g0`?

**Comment by jgreener64** [comment]
Thanks for the update.

It seems to work without error when I start Julia with `-g0`, the error shows with `-g1` and `-g2`.

**Comment by vchuravy** [comment]
SHould be fixed as of GPUCompiler 0.16.7, I just released a patch version of Enzyme that mandates that version.

**Comment by jgreener64** [comment]
Thanks, that worked for me.
